src/vm/exceptionhandling.cpp

```diff
@@ -3639,6 +3668,20 @@ ExceptionTracker* ExceptionTracker::GetOrCreateTracker(
                                            pExceptionRecord,
                                            pContextRecord);
 
+        if (fIsInterleavedHandling && fTransitionFromSecondToFirstPass)
+        {
+            // In interleaved exception handling a single exception is handled in an interleaved
+            // series of first and second passes. When we create a new tracker as a result
+            // of switching from the 2nd pass back to the 1st pass, we need to carry over
+            // several members that were set when processing one of the previous frames.
+            pNewTracker->m_sfResumeStackFrame = previousTrackerResumeStackFrame;
+            pNewTracker->m_uCatchToCallPC = previousTrackerCatchToCallPC;
+            pNewTracker->m_pClauseForCatchToken = previousTrackerClauseForCatchToken;
```

Comment:
These need to be copied over too. See my testing method below, the rethrow in the catch block. In this one, they are determined at the first frame being processed by the ProcessCLRException and nothing else updates them until they are used when calling the actual handler after all the frames are unwound. The rethrow inserts a bunch of native frames on the stack, so the unwinding needs to be done in the interleaved manner.
```csharp
    static void D()
    {
        try
        {
            try
            {
                try
                {
                    Console.WriteLine("Throwing an exception");
                    throw new Exception("Did you catch it in the right handler?");
                }
                catch (IndexOutOfRangeException e)
                {
                    Console.WriteLine("Test D: Failed. Called wrong handler.\n" + e.ToString());
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine("Test D: Caught exception!\n" + ex.ToString());
                throw;
            }
        }
        catch (Exception ex)
        {
            Console.WriteLine("Test D: success! Caught re-thrown exception!\n" + ex.ToString());
        }
    }
```